test/phoenix_live_view_test.exs

```diff
@@ -291,6 +306,29 @@ defmodule Phoenix.LiveViewUnitTest do
                __changed__: %{existing: true, notexisting: true, existing2: true}
              }
     end
+
+    test "has access to assigns" do
+      socket =
+        put_in(@socket.private[:assign_new], {%{existing: "existing-parent"}, []})
+        |> assign(existing2: "existing2")
+        |> assign_new(:existing, fn -> "new-existing" end)
+        |> assign_new(:existing_parent, fn %{existing: existing} -> existing end)
+        |> assign_new(:existing3, fn %{existing2: existing2} -> existing2 end)
+
+      assert socket.assigns == %{
+               existing: "new-existing",
+               existing_parent_updated: "new-existing",
+               existing3: "existing2",
+               live_action: nil,
+               flash: %{},
+               __changed__: %{
+                 existing: true,
+                 existing2: true,
+                 existing3: true,
+                 existing_parent: true
+               }
+             }
```

Comment:
Q: Are you intently ignoring in the case of `fn _ -> "new-existing" end)` to cause the fun/1 ?
I just assumed it would call the fun/0 and that you could drop the ignored.